web/src/reactions.js

```diff
@@ -248,10 +248,16 @@ export function add_reaction(event) {
 
     if (clean_reaction_object) {
         clean_reaction_object.user_ids.push(user_id);
-        const should_display_reactors = check_should_display_reactors(message.clean_reactions);
+        const reaction_counts_and_user_ids = get_reaction_counts_and_user_ids(message);
+        const should_display_reactors = check_should_display_reactors(reaction_counts_and_user_ids);
```

Comment:
This commit shouldn't make anything quadratic (as far as I can tell) because it's changing what used to happen in one loop, to two loops. Sometimes with appending an extra item to `reaction_counts_and_user_ids` in the middle.

Reply:
Agreed.